src/librustc_typeck/diagnostics.rs

```diff
@@ -1972,6 +1972,88 @@ To learn more about traits, take a look at the Book:
 https://doc.rust-lang.org/book/traits.html
 "##,
 
+E0174: r##"
+This error occurs because of unboxed closure explicitly methods calls is
+experimental in current Rust version.
+
+Example of erroneous code:
+
+```compile_fail
+ fn foo<F: Fn(&str)>(mut f: F) {
+     f.call(("call",));
+     //~ ERROR explicit use of unboxed closure method `call`
+     f.call_mut(("call_mut",));
+     //~ ERROR explicit use of unboxed closure method `call_mut`
+     f.call_once(("call_once",));
+     //~ ERROR explicit use of unboxed closure method `call_once`
+ }
+
+fn bar(name: &str) {
+    println!("Calling {} it works!", name);
+}
+
+fn main() {
+    foo(bar);
+}
+```
+
+Rust’s implementation of closures is a bit different than other languages.
+They are effectively syntax sugar for traits `Fn`, `FnMut` and `FnOnce`.
+To understand better how the closures are implemented see here:
+https://doc.rust-lang.org/book/closures.html#closure-implementation
+
+To fix it you can just call them using `()`, like `foo()`. Although, when you
+execute the closure with `()` under the hood you are executing the methods
+`call`, `call_mut` or `call_once` calling them explicitly is currently
```

Comment:
Yeah... When I read it again after did the PR I thought: "I am missing something there"

How about this:

> To fix it you can just call them using `()`, like `foo()`. When you execute the closure with `()`, under the hood you are executing the method `call`, `call_mut` or `call_once`. However, using them explicitly is currently an experimental feature.